### PR TITLE
Add option to crop preview images for Imagick

### PR DESCRIPTION
**Pull request by nvindice**
See https://github.com/aimeos/aimeos-core/commit/8efdd95507d26189c2573551e165dfc73964cc63

### CODE REVIEW

File: lib/mwlib/src/MW/Media/Image/Imagick.php [hunk 1]
```diff
@@ -157,29 +157,37 @@ public function save( string $filename = null, string $mimetype = null ) : ?stri
 	 *
 	 * @param int|null $width New width of the image or null for automatic calculation
 	 * @param int|null $height New height of the image or null for automatic calculation
-	 * @param bool $fit True to keep the width/height ratio of the image
+	 * @param int|null $fit 0 keeps image ratio, 1 enforces target size with scaling, 2 with cropping
 	 * @return \Aimeos\MW\Media\Image\Iface Self object for method chaining
 	 */
-	public function scale( int $width = null, int $height = null, bool $fit = true ) : \Aimeos\MW\Media\Image\Iface
+	public function scale( int $width = null, int $height = null, int $fit = null ) : \Aimeos\MW\Media\Image\Iface
```
Comment (aimeos) [inline lib/mwlib/src/MW/Media/Image/Imagick.php:163]: Why did you use `int|null` for the `$fit` parameter? NULL is no used value.

Comment (nvindice) [reply lib/mwlib/src/MW/Media/Image/Imagick.php]: You're right, I used it in a previous version. Cleaned it up.

**Comment by aimeos** [comment]
OK, looks good for Imagick and now you need to add the same feature to the Standard.php and SVG.php classes too so all support the same features again.

**Comment by nvindice** [comment]
Correct me if I'm wrong, but the Standard.php class always crops the image if $fit is set, right? So in fact the scaling algorithm from commit https://github.com/aimeos/aimeos-core/commit/8efdd95507d26189c2573551e165dfc73964cc63 is missing here? Same with SVG?

I don't know anything about the SVG library, so I would offer you to write the scaling algorithm for Standard.php if you write it for the SVG.php class.

**Comment by aimeos** [comment]
Yes, we can care about the SVG part

**Comment by nvindice** [comment]
The `resize()` algorithm in `Standard` copied from invalid source image areas (negative offset). I could not figure out why this would be a good idea so I fixed the code in order to add colored/transparent backgrounds.

**Comment by aimeos** [comment]
Thanks a lot!

**Pull request merged**